src/main/java/werkit/WerkIt.java

```diff
@@ -252,6 +280,43 @@ private void loadScheduleFile() throws IOException {
             logger.log(Level.WARNING, "Unknown file name was encountered.");
         }
 
+        if (!isScheduleFileLoadSuccessful) {
+            fileManager.deleteAndRecreateScheduleFile();
+            fileManager.rewriteAllDaysScheduleToFile(getDayList());
+            System.out.println("The corrupted schedule data has been removed.");
+            logger.log(Level.INFO, "Schedule file data loaded with corrupted data removed.");
+        }
+
         logger.log(Level.INFO, "Schedule file data loaded.");
     }
+
+    /**
+     * Reloads the schedule file to capture the effects in Daylist due to the deletion of workout and plan.
+     * @throws IOException  If the application is unable to open the schedule file.
+     */
+    private void reloadScheduleFile() throws IOException {
+        boolean isScheduleFileLoadSuccessful;
+        String[] schedule = dayList.getPrintSchedule();
+        this.dayList.clearAllSchedule();
+        String[] updatedSchedule;
+        for (int i = 0; i < schedule.length; i++) {
+            if (schedule[i] == null) {
+                schedule[i] = "rest day";
+            }
+        }
+        isScheduleFileLoadSuccessful = fileManager.reloadScheduleFromFile(getDayList());
+        if (!isScheduleFileLoadSuccessful) {
+            fileManager.deleteAndRecreateScheduleFile();
+            fileManager.rewriteAllDaysScheduleToFile(getDayList());
+            updatedSchedule = dayList.getPrintSchedule();
+            for (int i = 0; i < schedule.length; i++) {
+                if (!updatedSchedule[i].equals(schedule[i])) {
+                    var dayNum = i + 1;
+                    System.out.println("Schedule '" + schedule[i] + "' on " + dayList.covertDayNumberToDay(dayNum)
+                            + " has been removed.");
+                }
+            }
+            logger.log(Level.INFO, "Schedule file data updated due to deletion on workout/plan.");
+        }
+    }
```

Comment:
Since this is a file-related method, would it be for it to be in the `FileManager` class?

Reply:
Since it is a similar method to `loadScheduleFile()` , `loadPlanFile()` and `loadWorkoutFile()`, and they are all in WerkIt.java, I then put this method in WeriIt.java. The other reason is that we need to make use of the `getDayList()` to get the DayList, I assume that is the reason why you also put `loadScheduleFile()` , `loadPlanFile()` and `loadWorkoutFile()` in WerkIt instead of `FileManager`.

Reply:
Oh yeah, I forgot about that design consideration. Thanks for pointing it out! Then yeah can keep this method in the `WerkIt` class. Thank you :)